pelican/settings.py

```diff
@@ -660,3 +661,16 @@ def configure_settings(settings):
             continue            # setting not specified, nothing to do
 
     return settings
+
+
+def coerce_overrides(overrides):
+    coerced = {}
+    setting_types = {k: type(v) for k, v in DEFAULT_CONFIG.items()}
+    types_to_cast = {int, str}
```

Comment:
This is slightly tricky. Some settings allow `None` as well as a value (i.e. `SUMMARY_MAX_LENGTH` again... or `FEED_*` options). I don't have a clear solution for those.

Reply:
Instead of auto-coercing str/int I could make all overrides use JSON notation. I wanted to avoid this because you'd need to extra quote strings then
` overridesetting='"some-val"' `

Which seems a bit verbosey

Reply:
The problem is you can't use override to set these variables as `None` (for JSON it means `SETTING=null` I guess). We can consider that as a limitation I suppose and I won't be too bothered with it. 

For the `int` defaults, you could `try` to `int()` it and fallback to JSON if it errors, but string ones are hopeless since `'null'` is valid.

Reply:
Ok, I've added a second pass when getting a ValueError on int() casts